### ISSUE TITLE
Not able to run more than one query without restarting express server

### ISSUE THREAD

**Issue by TriStarGod** [tgriesser/knex#2346]
# Environment

Knex version: 0.14.1
pg: 7.4.0
Database + version: postgresql 9.6
OS: Windows 10 Pro x64

Select applicable tempalate from below.
If issue is about oracledb support tag @ atiertant. For MSSql tag @ Grimace1975 .
Rest of dialects doesn't need tags.

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do
Refer to https://stackoverflow.com/questions/47424341/unexpected-behavior-with-knexs-select
At first, I was adding new request type and I faced problems with Select. Now, I'm facing problems with all the request functions (worked previously).
2. Error message
No error message. It just returns undefined anytime the query is changed.
3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to mysql or postgresql, then single file example which initializes
   needed data and demonstrates the problem.

**Comment by elhigu** [comment]
Cannot reproduce. Please create the runnable test case that can be  executed by other people too.

**Comment by TriStarGod** [comment]
FInally got a runkit sample. Could not get a pg one working so using SQLIte.
https://runkit.com/tristargod/runkit-npm-knex

**Comment by elhigu** [comment]
tldr; Knex query builders are mutable so when re-using them .clone() is necessary.

nice example, from that it was easy to spot the problem 👍  

You are reusing the same query builder multiple times without cloning it between the queries. If you would run your code with DEBUG=knex:* environment variable set, you would see that constructed queries are not correct after the first call.

**Comment by TriStarGod** [comment]
@elhigu That makes sense. Didn't consider how queries were constructed by knex. 
In my example, instead of clone, I plan to use the following since its shorter.
```
const SQL = () => k('SomeObj');
// And my calls were turn into
SQL().where ....
```
Is that fine or would I take performance hit?

Also, by set environment variables, do you mean in Windows 10 x64, open System Properties > Advanced tab > Environment Variables > System variables > New > variable name: DEBUG / variable value: "knex:*" ?

I also tried setting the development config with debug: true and I saw additional output. Is that the same thing?

Thanks!

**Comment by elhigu** [comment]
Building the queries is pretty much never performance bottleneck. It is usually deserialising data read from DB. { debug: true } config parameter and that DEBUG environment variable are different things (I know that is really confusing). I would use git bash and set environment variable there. Or either way also in cmd just use commandline for setting them when running knex like `DEBUG=knex:* node myapp.js`

**Comment by TriStarGod** [comment]
Ok, got it. Thanks. #
